wgengine/wgengine.go

```diff
@@ -22,20 +25,25 @@ type PeerStatus struct {
 	NodeKey          tailcfg.NodeKey
 }
 
+// Status is the Engine status.
 type Status struct {
 	Peers      []PeerStatus
 	LocalAddrs []string // TODO(crawshaw): []wgcfg.Endpoint?
 }
 
-type StatusCallback func(s *Status, err error)
+// StatusCallback is the type of status callbacks used by Engine.SetStatusCallback.
+// TODO: exactly one of Status or error is set?
+type StatusCallback func(*Status, error)
 
+// RouteSettings is ... TODO.
```

Comment:
It's the full WireGuard config data (set of peers keys, IP, etc is `wgcfg.Config`) plus things that WireGuard doesn't actually do itself: the DNS servers to use.

I don't know why LocalAddr is there, and how it differs from what's inside wgcfg.Config.

Reply:
Thanks. Updated.